scripts/install

```diff
@@ -1,117 +1,170 @@
+# Create a system user
+ynh_system_user_create "$app"
 
-sudo yunohost app initdb $db_user -p $db_pwd
-sudo yunohost app setting $app mysqlpassword -v $db_pwd
+#=================================================
+# PHP-FPM CONFIGURATION
+#=================================================
 
-####################################################
-#   Nginx configuration                            #
-####################################################
+# Create a dedicated php-fpm config
+ynh_add_fpm_config
 
-sed -i "s@YNH_WWW_PATH@$path@g"         ../conf/nginx.conf
-sed -i "s@YNH_WWW_ALIAS@$final_path/@g" ../conf/nginx.conf
-sudo cp ../conf/nginx.conf /etc/nginx/conf.d/$domain.d/$app.conf
+#=================================================
+# GENERIC FINALIZATION
+#=================================================
+# SECURE FILES AND DIRECTORIES
+#=================================================
 
-####################################################
-#   SSOWat configuration                           #
-####################################################
+chown "$app"  "$final_path/config"
+chmod 750     "$final_path/config"
 
-if [ "$is_public" = "Yes" ];
+for folder in attachments cache exports files imports logs photos templates_c tempimages
+do
+    chown "$app"  "$final_path/data/$folder"
+    chmod 750     "$final_path/data/$folder"
+done
+
+#=================================================
+# SETUP LOGROTATE
+#=================================================
+
+# Use logrotate to manage application logfile(s)
+ynh_use_logrotate
+
+#=================================================
+# SETUP SSOWAT
+#=================================================
+
+if [[ "$path_url" == "/" ]]
+then
+	# ynh panel is only comptable with non-root installation
+	ynh_replace_string "  include conf.d/" "  #include conf.d/" "$finalnginxconf"
+else
+	ynh_replace_string "^#sub_path_only"   ""                   "$finalnginxconf"
+fi
+ynh_store_file_checksum "$finalnginxconf"
+
+if [ "$is_public" -eq 0 ]
+then	# Remove the public access
+	ynh_app_setting_delete "$app" skipped_uris
+fi
+# Make app public if necessary
+if [ "$is_public" -eq 1 ]
 then
-  # unprotected_uris allows SSO credentials to be passed anyway.
-  sudo yunohost app setting $app unprotected_uris -v "/"
+	# unprotected_uris allows SSO credentials to be passed anyway.
+	ynh_app_setting_set "$app" skipped_uris "/"
 fi
 
-####################################################
-#   Restart services                               #
-####################################################
+#=================================================
+# RELOAD NGINX
+#=================================================
+
+systemctl reload nginx
+
+#=================================================
+# SEND A README FOR THE ADMIN
+#=================================================
 
-sudo service nginx reload
-sudo yunohost app ssowatconf
+message="Galette need you to finish the installation manually.
```

Comment:
Meh, if the previous method was able to do this automatically, why make it manual with steps that involve the CLI ?

Reply:
yeah, it sucks :)
but it makes sure we respects the process and documentation of the upstream project, and it limits the amount of non-standard operation
once upstream support without those old-school stuff, we'll improve it :)

Reply:
Still I don't understand ? :/

How does the previous method does "not respect the process and documentation of the upstream project" (maybe it changed somehow since I worked on it, idk)

To me an app is supposed to automatize the whole install process and it currently does ... I don't see what makes it mandatory to un-automatize this, and especially our current UX of "reading logs in the webadmin" is not good (people often just see "The ssowat conf has been regenerated" and dont realize there are other lines) so I wouldn't hide a technical explanation of how to finish the install in there...

Sure using the whole python script thing to browse the pages and fill the forms is a "non-standard operation", but every app is non-standard anyway ...

Reply:
Here is my bug report: https://bugs.galette.eu/issues/1180